packages/app-earn-ui/src/components/molecules/ErrorMessage/ErrorMessage.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/consistent-indexed-object-style */
+import { type FC } from 'react'
+
+import { Icon, type IconNamesList } from '@/components/atoms/Icon/Icon'
+import { Text } from '@/components/atoms/Text/Text'
+
+import styles from './ErrorMessage.module.scss'
+
+type ErrorVariant = 'general' | 'critical' | 'warning' | 'success'
+
+interface ErrorMessageProps {
+  error: string
+  variant?: ErrorVariant
+  iconName?: IconNamesList
+}
+
+const variantToIconMap: Record<ErrorVariant, IconNamesList> = {
+  general: 'info',
+  critical: 'warning',
+  warning: 'warning',
+  success: 'checkmark',
+}
+
+const variantToColorMap: Record<ErrorVariant, string> = {
+  general: 'var(--color-background-interactive-bold)',
+  critical: 'var(--earn-protocol-critical-100)',
+  warning: 'var(--earn-protocol-warning-100)',
+  success: 'var(--earn-protocol-success-100)',
+}
+
+const variantToBackgroundMap: Record<ErrorVariant, string> = {
+  general: 'var(--gradient-earn-protocol-dark-10)',
+  critical: 'var(--earn-protocol-critical-10)',
+  warning: 'var(--earn-protocol-warning-10)',
+  success: 'var(--earn-protocol-success-10)',
```

Comment:
feels a little bit odd to have success variant within error component

Reply:
I agree. My rationale was because Anna/Hazel have called it that in the design system. Wanted to keep things aligned. I'll chat to Anna about it and get it renamed in both places.

![CleanShot 2025-02-24 at 09 38 46@2x](https://github.com/user-attachments/assets/fe65f189-54b8-45a1-9c91-75dc6df7c108)